context/index.tsx

```diff
@@ -24,6 +28,8 @@ const AppContext = createContext<AppSharedState>({} as AppSharedState);
 const AppSharedState = () => {
   const [loading, setLoading] = useState(true);
   const [dogs, setDogs] = useState([] as DogDetails[]);
+  const [likedDogs, setLikedDogs] = useState([] as DogDetails[]);
```

Comment:
useState accepts generic with the type of data

```suggestion
  const [likedDogs, setLikedDogs] = useState<DogDetails[]>([]);
```